assets/state-dra-driver/0700_daemonset.yaml

```diff
@@ -0,0 +1,130 @@
+      maxSurge: 0
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: nvidia-dra-driver-kubelet-plugin
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: nvidia-dra-driver
+      # TODO: revisit the affinity / nodeSelector for this daemonset
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: feature.node.kubernetes.io/pci-10de.present
+                operator: In
+                values:
+                - "true"
+            - matchExpressions:
+              - key: feature.node.kubernetes.io/cpu-model.vendor_id
+                operator: In
+                values:
+                - NVIDIA
+            - matchExpressions:
+              - key: nvidia.com/gpu.present
+                operator: In
+                values:
+                - "true"
+      initContainers:
+      - image: "FILLED BY THE OPERATOR"
+        name: driver-validation
+        command: [ 'sh', '-c' ]
+        args: [ "until [ -f /run/nvidia/validations/driver-ready ]; do echo waiting for driver to be setup; sleep 5; done" ]
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: run-nvidia-validations
+          mountPath: /run/nvidia/validations
+          mountPropagation: HostToContainer
+      containers:
+      - name: plugin
+        image: "FILLED BY THE OPERATOR"
+        imagePullPolicy: IfNotPresent
+        command: ["/bin/bash", "-c"]
+        args:
+          - /bin/entrypoint.sh
+        env:
+        - name: MASK_NVIDIA_DRIVER_PARAMS
+          value: "false"
+        - name: NVIDIA_VISIBLE_DEVICES
+          value: void
+        - name: CDI_ROOT
+          value: /var/run/cdi
+        - name: NVIDIA_MIG_CONFIG_DEVICES
+          value: all
+        - name: DEVICE_CLASSES
+          value: imex
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: nvidia-dra-driver-kubelet-plugin-entrypoint
+          readOnly: true
+          mountPath: /bin/entrypoint.sh
+          subPath: entrypoint.sh
+        - name: run-nvidia-validations
+          mountPath: /run/nvidia/validations
+        - name: driver-install-dir
+          mountPath: /driver-root
+          mountPropagation: HostToContainer
+          readOnly: true
+        - name: host-root
+          mountPath: /host
+          readOnly: true
+        - mountPath: /var/lib/kubelet/plugins_registry
+          name: plugins-registry
+        - mountPath: /var/lib/kubelet/plugins
+          mountPropagation: Bidirectional
```

Comment:
Question: Is bidirectional mount propagation really needed in the plugins folder?

Reply:
I am not sure. I copied this from the upstream DRA helm chart: https://github.com/NVIDIA/k8s-dra-driver/blob/32805fec62d6b3269e75ddb0eddeaa5c4d214564/deployments/helm/k8s-dra-driver/templates/kubeletplugin.yaml#L99

@klueska do you know if this is required?